File: iosVariant/iosVariant/Servers/DirectoryItemView.swift
/*
 * Variant - A digital comic book reading application for the iPad and Android tablets.
 * Copyright (C) 2025, The ComiXed Project
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses>
 */

import SwiftUI
import Variant

private let TAG = "DirectoryItemView"

struct DirectoryItemView: View {
    let entry: DirectoryEntry

    var onLoadDirectory: (String) -> Void

    var body: some View {
        VStack {
            Text(entry.title)
            Text(entry.filename)
        }
        .onTapGesture {
            Log().info(tag: TAG, message: "Directory clicked: \(entry.path)")
            onLoadDirectory(entry.path)
        }
    }
}

#Preview {
    DirectoryItemView(
        entry: DIRECTORY_LIST.filter { $0.isDirectory }.first!,
        onLoadDirectory: { _ in }
    )
}

File: androidVariant/src/main/java/org/comixedproject/variant/android/view/server/DirectoryItemView.kt
/*
 * Variant - A digital comic book reading application for the iPad and Android tablets.
 * Copyright (C) 2025, The ComiXed Project
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses>
 */

package org.comixedproject.variant.android.view.server

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.MoreVert
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.ElevatedCard
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.MaterialTheme.colorScheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import org.comixedproject.variant.android.DIRECTORY_LIST
import org.comixedproject.variant.android.VariantTheme
import org.comixedproject.variant.model.library.DirectoryEntry
import org.comixedproject.variant.platform.Log

private const val TAG = "DirectoryDetailView"

@Composable
fun DirectoryItemView(
    directoryEntry: DirectoryEntry,
    onLoadDirectory: (String) -> Unit,
    modifier: Modifier = Modifier
) {
    ElevatedCard(
        colors = CardDefaults.cardColors(containerColor = colorScheme.surface),
        modifier = modifier
            .fillMaxWidth()
    ) {
        Column(
            modifier = modifier
                .padding(16.dp)
                .fillMaxWidth()
                .clickable {
                    Log.info(TAG, "Loading path: ${directoryEntry.path}")
                    onLoadDirectory(directoryEntry.path)
                }
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.SpaceBetween,
                modifier = Modifier.fillMaxWidth()
            ) {
                Text(
                    text = "${directoryEntry.title}",
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.Bold,
                    textAlign = TextAlign.Left,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                    modifier = Modifier.weight(1f)
                )

                Icon(Icons.Default.MoreVert, contentDescription = directoryEntry.title)
            }
        }
    }
}

@Composable
@Preview
fun DirectoryItemViewPreview() {
    VariantTheme {
        DirectoryItemView(
            DIRECTORY_LIST.filter { it.isDirectory }.first(),
            onLoadDirectory = { _ -> })
    }
}